cairo/ethereum/cancun/blocks.cairo

```diff
@@ -0,0 +1,110 @@
+from ethereum.base_types import (
+    U64,
+    U256,
+    Bytes,
+    Bytes8,
+    Bytes32,
+    Uint,
+    TupleBytes32,
+    bool,
+    TupleBytes,
+)
+from ethereum.cancun.fork_types import Address, Bloom, Root
+from ethereum.crypto.hash import Hash32
+
+struct WithdrawalStruct {
+    index: U64,
+    validator_index: U64,
+    address: Address,
+    amount: U256,
+}
+
+struct Withdrawal {
+    value: WithdrawalStruct*,
+}
+
+struct TupleWithdrawalStruct {
+    value: Withdrawal,
+    len: felt,
+}
+
+struct TupleWithdrawal {
+    value: TupleWithdrawalStruct*,
+}
+
+struct HeaderStruct {
+    parent_hash: Hash32,
+    ommers_hash: Hash32,
+    coinbase: Address,
+    state_root: Root,
+    transactions_root: Root,
+    receipt_root: Root,
+    bloom: Bloom,
+    difficulty: Uint,
+    number: Uint,
+    gas_limit: Uint,
+    gas_used: Uint,
+    timestamp: U256,
+    extra_data: Bytes,
+    prev_randao: Bytes32,
+    nonce: Bytes8,
+    base_fee_per_gas: Uint,
+    withdrawals_root: Root,
+    blob_gas_used: U64,
+    excess_blob_gas: U64,
+    parent_beacon_block_root: Root,
+}
```

Comment:
A question on this; as now we pull all these types (like the Header here for example) to the Ethereum Execution Layer spec, how are we going to match "exactly" the reth types on the other side for the interaction with the ExEx? 

A few weeks ago, I remember that you made a Header Cairo struct that exactly matches the alloy type line by line and type by type, so that means here in Cairo that we will now have a Header Execution Spec struct and a Header Alloy struct or are we going to do the matching afterwards or via other means?

Reply:
good question, so for now I've not updated the main flow, this is like a new line of research (as we spoke) but I wanted to clean the wip and push it to merge. The `os.cairo` still uses the previously defined `Headers` and related types.

That being said, I think that the previous approach was less maintainable as we defined new types in cairo and then new types in python/rust to match them, so doing a eels/alloy to python/rust equivalent of cairo conversion, then this python/rust type to cairo. I think that it's better to skip these new types and do the job to "use directly" the eels ones. If you implement the same `gen_arg` in rust as in python, because eels and alloy types are supposed to be the same, you should be good, and it requires less per final type logic